napari/layers/surface/surface.py

```diff
@@ -309,6 +312,10 @@ def shading(self):
 
     @shading.setter
     def shading(self, shading):
+        if self._ndisplay < 3:
+            raise ValueError(
```

Comment:
Can we maybe make this a warning and then just overwrite `self._shading = 'none'`. I feel like i'd rather just warn and fix it then raise an error